pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,500 @@
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('D',0),('D',1)),(('I',0),('I',1)))])
+        >>> dbn.inter_slice(1)
+        [edge_type(start=('D', 0), end=('D', 1)),
+        edge_type(start=('I', 0), end=('I', 1))]
+        """
+        if not isinstance(timeslice, int):
+            raise ValueError('the timeslice is not an integer value')
+        if timeslice < 0:
+            raise ValueError('negative values for a timeslice are not permissible')
+        if timeslice > self.timestate:
+            raise ValueError('the bayesian network does not contain the inter slice edges of this timeslice')
+        return list(filter(lambda x: x.end[1] == timeslice, self.inter_edges))
+
+    def add_dbn_cpds(self, timeslice, *cpds):
+        """
+        Add CPD (Conditional Probability Distribution) to the Bayesian Model.
+
+        Parameters
+        ----------
+        cpds  :  list, set, tuple (array-like)
+            List of cpds (TabularCPD, TreeCPD, RuleCPD, Factor)
+            which will be associated with the model
+            
+        timeslice :integer value. should be ranging from 0
+                   to the number of the slices present in
+                   the bayesian network.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> from pgmpy.factors.CPD import TabularCPD
+        >>> student = DBN('D','G','I','S','L')
+        >>> grades_cpd = TabularCPD('G', 3, [[0.1,0.1,0.1,0.1,0.1,0.1],
+        ...                                  [0.1,0.1,0.1,0.1,0.1,0.1],
+        ...                                  [0.8,0.8,0.8,0.8,0.8,0.8]],
+        ...                         evidence=['D', 'I'], evidence_card=[2, 3])
+        >>> student.add_dbn_cpds(0, grades_cpd)
+
+        +------+-----------------------+---------------------+
+        |diff: |          easy         |         hard        |
+        +------+------+------+---------+------+------+-------+
+        |intel:| dumb |  avg |  smart  | dumb | avg  | smart |
+        +------+------+------+---------+------+------+-------+
+        |gradeA| 0.1  | 0.1  |   0.1   |  0.1 |  0.1 |   0.1 |
+        +------+------+------+---------+------+------+-------+
+        |gradeB| 0.1  | 0.1  |   0.1   |  0.1 |  0.1 |   0.1 |
+        +------+------+------+---------+------+------+-------+
+        |gradeC| 0.8  | 0.8  |   0.8   |  0.8 |  0.8 |   0.8 |
+        +------+------+------+---------+------+------+-------+
+        """
+        for cpd in cpds:
+            if not isinstance(cpd, (TabularCPD, TreeCPD, RuleCPD)):
+                raise ValueError('Only TabularCPD, TreeCPD or RuleCPD can be'
+                                 ' added.')
+
+            if (cpd.variable, timeslice) not in self.dbn_cpds:
+                self.dbn_cpds[(cpd.variable, timeslice)] = cpd
+
+    def get_dbn_cpds(self, node=None, timeslice=None):
```

Comment:
Also please have a look at the ```get_cpds``` method in ```BayesianModel``` to have an idea what it should return and how the API should be structured.